zookeeper-api/src/main/java/org/apache/helix/zookeeper/zkclient/ZkClient.java

```diff
@@ -185,6 +188,7 @@ protected ZkClient(IZkConnection zkConnection, int connectionTimeout, long opera
     if (zkConnection == null) {
       throw new NullPointerException("Zookeeper connection is null!");
     }
+    _syncOnNewSession = true;
```

Comment:
You mentioned we will have more changes here. Will we make it optional? If so, I think we should have the change in this PR.

Reply:
I would say, let us don't make it optional. This is across session "sequential consistency"

@pkuwm, what is your take?

Reply:
It is worth making it configurable. For all Helix related component, we can turn it on. Otherwise, every user operation can be delay a little bit for sync. It changed our current behavior for user.

Reply:
This is a good idea.

Reply:
Good point. Will add a system property then.